packages/block-library/src/heading/autogenerate-anchors.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { deburr, trim } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Runs a callback over all blocks, including nested blocks.
+ *
+ * @param {Object[]} blocks   The blocks.
+ * @param {Function} callback The callback.
+ *
+ * @return {void}
+ */
+const recurseOverBlocks = ( blocks, callback ) => {
+	for ( const block of blocks ) {
+		// eslint-disable-next-line callback-return
+		callback( block );
```

Comment:
nit: how about making this into a generator (`yield block`)? Then we could map a list of blocks later on, getAllHeadingAnchors would be just a `recursivelyGetBlocks( blockList ).filter(/*...*/).map(block => block.attributes.anchor)`

Reply:
This implementation was removed with the refactor in commit https://github.com/WordPress/gutenberg/pull/30825/commits/5bb9302bb8bc3f0f87ad793c25bfea38fe2e6ae2 so it's no longer an issue :+1: